### PR TITLE
Addon search with filter by multiple guid

### LINKED ISSUE #7119: Add an API method to get details about multiple addons with a single request
Over at https://bugzil.la/1402064 we are working on migrating Firefox from API version 1.5 to the new API.  Firefox currently uses the search API endpoint which can return the details for a list of addons (specific by guid) in a single request.  But there does not appear to be a way to do this with the new API.
The next Firefox ESR will be based on 59, which uplifts on Jan 22 so if we can get this all in place by then, the old API can be retired some time after ESR52 is end-of-life, otherwise we'll have to wait for ESR59 to reach end-of-life, which would be sometime in 2019 (!!)

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #7119

### CODE REVIEW

File: src/olympia/addons/tests/test_views.py [hunk 1]
```diff
@@ -2731,6 +2731,41 @@ def test_filter_by_multiple_authors(self):
...
+            self.url, {'guid': u'random@guid,random2@guid'})
+        assert data['count'] == 2
+        assert len(data['results']) == 2
+
+        result = data['results'][0]
+        assert result['id'] == addon.pk
+        assert result['slug'] == addon.slug
+        result = data['results'][1]
+        assert result['id'] == addon2.pk
+        assert result['slug'] == addon2.slug
```
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_views.py]: Can we add some random other guid string to the search string to test the behavior of invalid guids?

Comment (eviljeff) [reply src/olympia/addons/tests/test_views.py]: What kind of invalid guid and to test what exactly?  If it's an invalid guid it won't be found, like any other guid we don't have in the database.

Comment (EnTeQuAk) [reply src/olympia/addons/tests/test_views.py]: Right, that means if you're adding a guid that won't be found to line 2757 the assertion in 2760 will still be `2` which is what I want to see basically

Comment (EnTeQuAk) [reply src/olympia/addons/tests/test_views.py]: We don't have a "not found" test for guids for that filter as far as I can tell

**Pull request merged**